File: app/java/src/main/java/com/example/production/domain/model/inventory/StocktakingDetail.java
package com.example.production.domain.model.inventory;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 棚卸明細データエンティティ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StocktakingDetail {
    private Long id;
    private String stocktakingNumber;
    private Integer lineNumber;
    private String itemCode;
    private BigDecimal bookQuantity;
    private BigDecimal actualQuantity;
    private BigDecimal differenceQuantity;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Inventory/StocktakingDetail.cs
namespace ProductionManagement.Domain.Models.Inventory;

/// <summary>
/// 棚卸明細データ
/// </summary>
public class StocktakingDetail
{
    public long? Id { get; init; }
    public required string StocktakingNumber { get; init; }
    public required int LineNumber { get; init; }
    public required string ItemCode { get; init; }
    public required decimal BookQuantity { get; init; }
    public decimal? ActualQuantity { get; init; }
    public decimal? DifferenceQuantity { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }
}
